connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundIntermediateConnectorContextImpl.java

```diff
@@ -21,26 +21,28 @@
 import io.camunda.connector.api.inbound.InboundConnectorDefinition;
 import io.camunda.connector.api.inbound.InboundConnectorResult;
 import io.camunda.connector.api.inbound.InboundIntermediateConnectorContext;
-import io.camunda.connector.api.inbound.operate.ProcessInstance;
-import java.util.List;
+import io.camunda.connector.api.inbound.operate.ProcessInstanceContext;
 import java.util.Map;
-import java.util.function.Supplier;
 
+/**
+ * Implementation of {@link InboundIntermediateConnectorContext} that delegates operations to an
+ * underlying {@link InboundConnectorContext} and {@link ProcessInstanceContext}.
+ */
 public class InboundIntermediateConnectorContextImpl
     implements InboundIntermediateConnectorContext {
   private final InboundConnectorContext inboundContext;
-  private final Supplier<List<ProcessInstance>> processInstancesSupplier;
+  private final ProcessInstanceContext processInstanceContext;
```

Comment:
The idea was that a `ProcessInstanceContext` represents a processs instance context and not all process instances. A process instance context is the combination of a processs instance and the variables of the process instance + the connector inbound context.